File: Backend/fairshareapi/src/main/java/com/priyanshparekh/fairshareapi/balanceinfo/BalanceInfo.java
package com.priyanshparekh.fairshareapi.balanceinfo;

import jakarta.persistence.*;
import lombok.*;

@Setter
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "balance_info")
public class BalanceInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long groupId;
    private Long userId;
    @Enumerated(EnumType.STRING)
    private Direction direction;
    private Long otherUserId;
    private Double amount;

    protected enum Direction {
        OWES_TO,
        RECEIVES_FROM
    }

}

File: Frontend/FairShare/app/src/main/java/com/priyanshparekh/fairshare/model/BalanceInfo.kt
package com.priyanshparekh.fairshare.model

data class BalanceInfo(
    val id: Long?,
    val groupId: Long,
    val userId: Long,
    val direction: Direction,
    val otherUserId: Long,
    val amount: Double
)

enum class Direction {
    OWES_TO,
    RECEIVES_FROM
}

